web/src/stream_create.js

```diff
@@ -86,8 +96,18 @@ class StreamNameError {
         // out it already exists, and I was just too lazy to look at
         // the public streams that I'm not subscribed to yet.  Once I
         // realize the stream already exists, I may want to cancel.)
-        if (stream_name && stream_data.get_sub(stream_name)) {
-            this.report_already_exists();
+        const stream = stream_data.get_sub(stream_name);
+        if (stream_name && stream) {
+            if (stream.archived) {
+                this.report_already_exists(
+                    $t({defaultMessage: "An archived stream with this name already exists"}),
+                );
+                if (stream_settings_data.get_sub_for_settings(stream).can_change_name_description) {
+                    this.rename_archived_stream(stream.name);
+                }
+            } else {
+                this.report_already_exists();
+            }
```

Comment:
We probably should check that you have permission to administer the stream in question?

Reply:
I am not sure I fully understand.
`can_change_name_description` makes sure admin has the access to rename the stream.
And if someone with the permission to create streams attempts to create a stream with the same name as an existing one, they will receive an error message stating:

> A stream with this name already exists

Why is it necessary to check permissions before showing this error? If a user has the ability to create a stream and is attempting to create one with a name already in use, they should be informed that the name is already taken.